File: solutions/2026-08-06/javascript/2026-08-06-smallest-divisible-digit-product-i.js
// Summary: Find the smallest number >= n where the product of its digits is divisible by t.
// Link: https://leetcode.com/problems/smallest-divisible-digit-product-i/
// Approach:
// We need to iterate through numbers starting from n and check if the product of their digits is divisible by t.
// The first number that satisfies this condition will be our answer.
// To calculate the digit product, we can convert the number to a string, iterate through its characters (digits),
// convert each character back to a number, and multiply them. Special case: if any digit is 0, the product is 0,
// which is divisible by any t (since t >= 1).
// Time Complexity: O( (ans - n) * log10(ans) ). In the worst case, `ans` can be significantly larger than `n`.
// However, given the constraints (n <= 100, t <= 10), the search space is small. The maximum number we might
// check won't be excessively large for these constraints. For larger constraints, this approach could be slow.
// Space Complexity: O(log10(ans)) due to string conversion for digit product calculation.

/**
 * @param {number} n
 * @param {number} t
 * @return {number}
 */
const smallestDivisibleDigitProduct = (n, t) => {
    // Helper function to calculate the product of digits of a number.
    const getDigitProduct = (num) => {
        // If the number is 0, its digit product is 0.
        if (num === 0) {
            return 0;
        }
        let product = 1;
        // Convert the number to a string to easily access its digits.
        const numStr = String(num);
        // Iterate through each character of the string.
        for (let i = 0; i < numStr.length; i++) {
            const digit = parseInt(numStr[i], 10);
            // If any digit is 0, the entire product will be 0.
            if (digit === 0) {
                return 0;
            }
            // Multiply the current product by the digit.
            product *= digit;
        }
        // Return the calculated product.
        return product;
    };

    // Start checking from the given number 'n'.
    let currentNum = n;
    // Loop indefinitely until we find a number that satisfies the condition.
    while (true) {
        // Calculate the product of digits for the current number.
        const digitProduct = getDigitProduct(currentNum);
        // Check if the digit product is divisible by 't'.
        // The modulo operator (%) returns 0 if it's divisible.
        if (digitProduct % t === 0) {
            // If divisible, this is the smallest number >= n that satisfies the condition.
            return currentNum;
        }
        // If not divisible, increment the number and check the next one.
        currentNum++;
    }
};

File: solutions/2026-08-06/python/2026-08-06-smallest-divisible-digit-product-i.py
```python
# Finds the smallest number >= n whose digit product is divisible by t.
# Link: https://leetcode.com/problems/smallest-divisible-digit-product-i/
#
# Approach:
# We can iterate through numbers starting from n upwards. For each number,
# we calculate the product of its digits. If this product is divisible by t,
# we return the current number as it's the smallest such number encountered.
# The digit product calculation involves converting the number to a string,
# iterating through its characters, converting them back to integers, and
# multiplying them. Special handling for the digit 0 is needed, as any
# product involving 0 will be 0, which is divisible by any non-zero t.
#
# Time complexity: O(N * log10(N)), where N is the smallest number >= n satisfying the condition.
# In the worst case, we might iterate up to a certain point. The log10(N) factor comes
# from calculating the digit product of each number, which takes time proportional
# to the number of digits. Given the constraints on n (<= 100) and t (<= 10),
# the number of iterations and the magnitude of the numbers will be relatively small.
#
# Space complexity: O(log10(N)) due to string conversion for digit extraction.

class Solution:
    def smallestDivisibleDigitProduct(self, n: int, t: int) -> int:
        """
        Finds the smallest number greater than or equal to n such that the
        product of its digits is divisible by t.
        """

        # Iterate through numbers starting from n
        current_num = n
        while True:
            # Calculate the product of digits for the current number
            product = 1
            has_zero = False
            # Convert the number to a string to easily access its digits
            num_str = str(current_num)

            # Iterate through each character (digit) in the string
            for digit_char in num_str:
                digit = int(digit_char)
                # If a digit is 0, the product will be 0.
                # 0 is divisible by any non-zero t.
                if digit == 0:
                    has_zero = True
                    break # No need to multiply further if a zero is encountered
                product *= digit

            # Check if the product of digits is divisible by t
            # If has_zero is True, the product is effectively 0.
            # If t is 0, the problem statement constraints (t >= 1) prevent this.
            if has_zero or product % t == 0:
                return current_num # Found the smallest number satisfying the condition

            # If the condition is not met, increment the number and try again
            current_num += 1

```